botocore/paginate.py

```diff
@@ -189,14 +189,18 @@ def _inject_token_into_kwargs(self, op_kwargs, next_token):
 
     def _handle_first_request(self, parsed, primary_result_key,
                               starting_truncation):
-        # First we need to slice into the array and only return
-        # the truncated amount.
+        # If the payload is an array or string, we need to slice into it
+        # and only return the truncated amount.
         starting_truncation = self._parse_starting_token()[1]
         all_data = primary_result_key.search(parsed)
+        if isinstance(all_data, (list, string_types)):
```

Comment:
Is this right?  Isn't this equivalent to: `isinstance(all_data, (list, (basestring,)))` in python2?  Shouldn't it be `isinstance(all_data, (list, bastring))`?